File: src/main/java/com/multicampus/gamesungcoding/a11ymarketserver/feature/seller/dto/SellerApplyResponse.java
package com.multicampus.gamesungcoding.a11ymarketserver.feature.seller.dto;

import com.multicampus.gamesungcoding.a11ymarketserver.feature.seller.entity.Seller;
import com.multicampus.gamesungcoding.a11ymarketserver.feature.seller.entity.SellerGrades;
import com.multicampus.gamesungcoding.a11ymarketserver.feature.seller.entity.SellerSubmitStatus;

import java.time.LocalDateTime;
import java.util.UUID;

public record SellerApplyResponse(
        UUID sellerId,
        String sellerName,
        String userName,
        String userEmail,
        String userPhone,
        String businessNumber,
        SellerGrades sellerGrade,
        String sellerIntro,
        Boolean a11yGuarantee,
        SellerSubmitStatus sellerSubmitStatus,
        LocalDateTime submitDate,
        LocalDateTime approvedDate) {

    public static SellerApplyResponse fromEntity(Seller seller) {
        return new SellerApplyResponse(
                seller.getSellerId(),
                seller.getSellerName(),
                seller.getUser().getUserName(),
                seller.getUser().getUserEmail(),
                seller.getUser().getUserPhone(),
                seller.getBusinessNumber(),
                seller.getSellerGrade(),
                seller.getSellerIntro(),
                seller.getIsA11yGuarantee(),
                seller.getSellerSubmitStatus(),
                seller.getSubmitDate(),
                seller.getApprovedDate()
        );
    }

}

File: src/main/kotlin/com/multicampus/gamesungcoding/a11ymarketserver/feature/seller/dto/SellerApplyResponse.kt
package com.multicampus.gamesungcoding.a11ymarketserver.feature.seller.dto

import com.multicampus.gamesungcoding.a11ymarketserver.feature.seller.entity.SellerGrades
import com.multicampus.gamesungcoding.a11ymarketserver.feature.seller.entity.SellerSubmitStatus
import java.time.LocalDateTime
import java.util.UUID

data class SellerApplyResponse(
    val sellerId: UUID,
    val sellerName: String,
    val userName: String,
    val userEmail: String,
    val userPhone: String,
    val businessNumber: String,
    val sellerGrade: SellerGrades,
    val sellerIntro: String?,
    val a11yGuarantee: Boolean,
    val sellerSubmitStatus: SellerSubmitStatus,
    val submitDate: LocalDateTime,
    val approvedDate: LocalDateTime?
)
